Describe teams in data model (#64)
appuio/appuio-io-docs: docs/modules/ROOT/pages/references/architecture/identity-data-model.adoc

Organizations::
Each {product} organization is represented by a group object in Keycloak.

> Teams::
> Each team of an {product} organization is represented by a group object in Keycloak.
> A group object set below an organization group object in the group hierarchy becomes a team.

Reviewer: In #63 we're already using the word "represented" when talking about CRD/specs. Could be misleading here.
```suggestion
Teams::
Each team of an {product} organization is implemented as a group object in Keycloak.
A group object set below an organization group object in the group hierarchy becomes a team.
```

Author: Thanks for the input 😊
